src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -545,43 +545,39 @@ private boolean offerNotEnoughResources(Offer offer, double cpus, int mem) {
   }
 
   private void reconcileTasks(SchedulerDriver driver) {
-    updateReconciliationDeadline();
     driver.reconcileTasks(Collections.<Protos.TaskStatus> emptyList());
-    reconcilePersistentState();
-    reconciliationCompleted = true;
+    Timer timer = new Timer();
+    timer.schedule(new ReconcileStateTask(), conf.getReconciliationTimeout() * 1000);
```

Comment:
Sorry, I was incorrectly thinking that we were already doing periodic retries with (exponential) backoff, but this timeout is just saying that after a single reconcileTasks call we'll assume reconciliation is done. That means we won't get reconciliation status updates from any slave that wasn't already registered when the framework happened to reregister. When we're testing reconciliation, the order of reregistration of frameworks/slaves matters. When just the scheduler fails over we're fine, but if the master fails over, things get more complicated.